fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -61,7 +61,7 @@ def fit(self, X, y, aux_data=None, **kwargs):
             number_of_lagrange_multipliers = kwargs[self._KW_NUMBER_LAGRANGE_MULTIPLIERS]
 
         # Extract the target attribute
-        A = self._make_vector(aux_data, "aux_data")
+        A = self._make_vector(self.constraint.extract_A(X, aux_data), "A")
```

Comment:
This is one key part; we use the moment object to get the protected column which GridSearch requires

Reply:
What if it's not a dataframe?

Reply:
As noted in the title, this is incomplete. Dataframes were the easiest to get working, so I did those.